# INDIAN STARTUP FUNDING

## Introduction

**Project Objective**

The objective of this project is to analyze Indian startup funding data from 2015 to 2017 to understand funding trends, popular industries, major investors, and key startup hubs in India.

**Project Definition**

This project analyzes a dataset containing funding details of Indian startups, including funding date, city, investors, industry, and amount invested. The analysis focuses on identifying patterns in startup funding over time, the most funded sectors and cities, and the role of major investors in the Indian startup ecosystem.

**Dataset**

This dataset contains funding information for Indian startups from January 2015 to August 2017. It includes details such as the funding date, the city where the startup is based, the names of the investors, and the amount invested (in USD).

**Libraries**

First of all, we import the necessary Python libraries for data manipulation, numerical computations, and data visualization. These libraries provide essential tools for exploring, cleaning, and visualizing the dataset efficiently. 

- **pandas** is used for data loading and manipulation.
- **numpy** supports numerical operations.
- **matplotlib** and **seaborn** are used for data visualization and plotting.

In [1]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [2]:
import pandas as pd

In [3]:
import numpy as np

In [4]:
import matplotlib.pyplot as plt

In [5]:
pip install seaborn

Note: you may need to restart the kernel to use updated packages.


In [6]:
import seaborn as sns
sns.set_theme()   #it makes plots look nicer by default

## 1. Data Loading and Initial Overview

### 1.1 Loading the Dataset

We start by loading the dataset using **pandas**. 

In [7]:
df = pd.read_csv("Indian Startup Funding.csv")
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039                    ?                                      ?   
3040                    ?                                      ?   
3041                    ?                                      ?   
3042                    ?                                      ?   
3043                    ?                                      ?   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039              ?        Asia Pacific Internet Group        Private Equity   
3040              ?                     KARSEMVEN Fund        Private Equity   
3041              ?     Exfinity Fund, GrowX Ventures.        Private Equity   
3042              ?                         MakeMyTrip        Private Equity   
3043              ?  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks  
0     20,00,00,000                                  ?  
1        80,48,394                                  ?  
2      1,83,58,860                                  ?  
3        30,00,000                                  ?  
4        18,00,000                                  ?  
...            ...                                ...  
3039     45,00,000                                  ?  
3040      8,25,000                Govt backed VC Fund  
3041     15,00,000                                  ?  
3042             ?  Strategic Funding, Minority stake  
3043      1,40,000                                  ?  

[3044 rows x 10 columns]

### 1.2  Initial Overview

Now we can provide an overview of the dataset. 

**1.2.(a) Number of rows and columns**

By using **df.shape**, it returns a tuple: (rows, columns). Then we extract the number of rows and columns from it. 

In [8]:
print(df.shape)
print("Rows:",df.shape[0])
print("Columns:",df.shape[1])

(3044, 10)
Rows: 3044
Columns: 10


We can see that the dataset has 3044 rows and 10 columns. 

**1.2.(b) Data types of each column**

Using **df.dtypes**, we get

In [9]:
print(df.dtypes)

Sr_No                 int64
Date_dd/mm/yyyy      object
Startup_Name         object
Industry_Vertical    object
SubVertical          object
City__Location       object
Investors_Name       object
InvestmentnType      object
Amount_in_USD        object
Remarks              object
dtype: object


**1.2.(c) Initial observations**

Now we look onto some initial observations using **head()**, **info()** & **describe()**. 

- **head()** shows the first 5 rows of your dataset (default).
- **info()** provides summary information about the dataset.
- **describe()** provides statistical summary for numeric columns by default. 

In [10]:
df.head()

Sr_No Date_dd/mm/yyyy                  Startup_Name    Industry_Vertical  \
0      1      09-01-2020                      BYJU’S               E-Tech   
1      2      13-01-2020                        Shuttl       Transportation   
2      3      09-01-2020                     Mamaearth           E-commerce   
3      4      02-01-2020  https://www.wealthbucket.in/              FinTech   
4      5      02-01-2020                        Fashor  Fashion and Apparel   

                             SubVertical City__Location  \
0                             E-learning      Bengaluru   
1              App based shuttle service        Gurgaon   
2  Retailer of baby and toddler products      Bengaluru   
3                      Online Investment      New Delhi   
4            Embroiled Clothes For Women         Mumbai   

              Investors_Name       InvestmentnType Amount_in_USD Remarks  
0    Tiger Global Management  Private Equity Round  20,00,00,000       ?  
1  Susquehanna Growth Equity              Series C     80,48,394       ?  
2      Sequoia Capital India              Series B   1,83,58,860       ?  
3             Vinod Khatumal          Pre-series A     30,00,000       ?  
4    Sprout Venture Partners            Seed Round     18,00,000       ?

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3044 entries, 0 to 3043
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Sr_No              3044 non-null   int64 
 1   Date_dd/mm/yyyy    3044 non-null   object
 2   Startup_Name       3044 non-null   object
 3   Industry_Vertical  3044 non-null   object
 4   SubVertical        3044 non-null   object
 5   City__Location     3044 non-null   object
 6   Investors_Name     3044 non-null   object
 7   InvestmentnType    3044 non-null   object
 8   Amount_in_USD      3040 non-null   object
 9   Remarks            3044 non-null   object
dtypes: int64(1), object(9)
memory usage: 237.9+ KB


In [12]:
df.describe()

Sr_No
count  3044.000000
mean   1522.500000
std     878.871435
min       1.000000
25%     761.750000
50%    1522.500000
75%    2283.250000
max    3044.000000

## 2. Data Pre-processing 

Now we perform all the necessary cleaning steps. 

### 2.1 Removing Duplicates

First we look for any duplicate rows, using **df.duplicated()**. 

In [13]:
duplicates = df.duplicated()
print("Total number of duplicate rows:", duplicates.sum())  # total number of duplicate rows

Total number of duplicate rows: 0


We found 0 duplicate rows, so the dataset now contains only unique entries. 

### 2.2 Correcting Data Types 

Earlier in **1.2.(b)** we found that the data type of the column "Amount_in_USD" as object. Before we convert it into as float, if we have commas, currency symbols, spaces,...etc, cleaning it first helps a lot. 

In [14]:
df["Amount_in_USD_Clean"] = (
    df["Amount_in_USD"]
      .astype(str)
      .str.replace(",", "", regex=False)
      .str.replace("$", "", regex=False)
      .str.strip()
)

In [15]:
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039                    ?                                      ?   
3040                    ?                                      ?   
3041                    ?                                      ?   
3042                    ?                                      ?   
3043                    ?                                      ?   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039              ?        Asia Pacific Internet Group        Private Equity   
3040              ?                     KARSEMVEN Fund        Private Equity   
3041              ?     Exfinity Fund, GrowX Ventures.        Private Equity   
3042              ?                         MakeMyTrip        Private Equity   
3043              ?  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks Amount_in_USD_Clean  
0     20,00,00,000                                  ?           200000000  
1        80,48,394                                  ?             8048394  
2      1,83,58,860                                  ?            18358860  
3        30,00,000                                  ?             3000000  
4        18,00,000                                  ?             1800000  
...            ...                                ...                 ...  
3039     45,00,000                                  ?             4500000  
3040      8,25,000                Govt backed VC Fund              825000  
3041     15,00,000                                  ?             1500000  
3042             ?  Strategic Funding, Minority stake                   ?  
3043      1,40,000                                  ?              140000  

[3044 rows x 11 columns]

Now we convert it into float. 

In [16]:
df["Amount_in_USD_Clean"] = pd.to_numeric(df["Amount_in_USD_Clean"], errors="coerce")   # converts ?, empty strings, bad text into NaN

In [17]:
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039                    ?                                      ?   
3040                    ?                                      ?   
3041                    ?                                      ?   
3042                    ?                                      ?   
3043                    ?                                      ?   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039              ?        Asia Pacific Internet Group        Private Equity   
3040              ?                     KARSEMVEN Fund        Private Equity   
3041              ?     Exfinity Fund, GrowX Ventures.        Private Equity   
3042              ?                         MakeMyTrip        Private Equity   
3043              ?  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks  Amount_in_USD_Clean  
0     20,00,00,000                                  ?          200000000.0  
1        80,48,394                                  ?            8048394.0  
2      1,83,58,860                                  ?           18358860.0  
3        30,00,000                                  ?            3000000.0  
4        18,00,000                                  ?            1800000.0  
...            ...                                ...                  ...  
3039     45,00,000                                  ?            4500000.0  
3040      8,25,000                Govt backed VC Fund             825000.0  
3041     15,00,000                                  ?            1500000.0  
3042             ?  Strategic Funding, Minority stake                  NaN  
3043      1,40,000                                  ?             140000.0  

[3044 rows x 11 columns]

Now we move to the column "Date_dd/mm/yyyy". 

In [18]:
df["Date"] = (
    df["Date_dd/mm/yyyy"]
      .astype(str)
      .str.strip()                 # remove leading/trailing spaces
)

df["Date"] = pd.to_datetime(df["Date"], dayfirst=True, errors="coerce")

In [19]:
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039                    ?                                      ?   
3040                    ?                                      ?   
3041                    ?                                      ?   
3042                    ?                                      ?   
3043                    ?                                      ?   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039              ?        Asia Pacific Internet Group        Private Equity   
3040              ?                     KARSEMVEN Fund        Private Equity   
3041              ?     Exfinity Fund, GrowX Ventures.        Private Equity   
3042              ?                         MakeMyTrip        Private Equity   
3043              ?  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks  Amount_in_USD_Clean  \
0     20,00,00,000                                  ?          200000000.0   
1        80,48,394                                  ?            8048394.0   
2      1,83,58,860                                  ?           18358860.0   
3        30,00,000                                  ?            3000000.0   
4        18,00,000                                  ?            1800000.0   
...            ...                                ...                  ...   
3039     45,00,000                                  ?            4500000.0   
3040      8,25,000                Govt backed VC Fund             825000.0   
3041     15,00,000                                  ?            1500000.0   
3042             ?  Strategic Funding, Minority stake                  NaN   
3043      1,40,000                                  ?             140000.0   

           Date  
0    2020-01-09  
1    2020-01-13  
2    2020-01-09  
3    2020-01-02  
4    2020-01-02  
...         ...  
3039 2015-01-29  
3040 2015-01-29  
3041 2015-01-30  
3042 2015-01-30  
3043 2015-01-31  

[3044 rows x 12 columns]

In [20]:
df.dtypes

Sr_No                           int64
Date_dd/mm/yyyy                object
Startup_Name                   object
Industry_Vertical              object
SubVertical                    object
City__Location                 object
Investors_Name                 object
InvestmentnType                object
Amount_in_USD                  object
Remarks                        object
Amount_in_USD_Clean           float64
Date                   datetime64[ns]
dtype: object

By comparing the data types of the columns "Date_dd/mm/yyyy" and "Amount_in_USD" to "Date" and "Amount_in_USD_Clean" respectively, you can see the dfference. 

### 2.3 Handling Missing Values

Now we replace misiing values in the column "Amount_in_USD_Clean" with avaerage, using **fillna()**. 

In [21]:
df["Amount_in_USD_Clean"] = df["Amount_in_USD_Clean"].fillna(df["Amount_in_USD_Clean"].mean())


In [22]:
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039                    ?                                      ?   
3040                    ?                                      ?   
3041                    ?                                      ?   
3042                    ?                                      ?   
3043                    ?                                      ?   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039              ?        Asia Pacific Internet Group        Private Equity   
3040              ?                     KARSEMVEN Fund        Private Equity   
3041              ?     Exfinity Fund, GrowX Ventures.        Private Equity   
3042              ?                         MakeMyTrip        Private Equity   
3043              ?  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks  Amount_in_USD_Clean  \
0     20,00,00,000                                  ?         2.000000e+08   
1        80,48,394                                  ?         8.048394e+06   
2      1,83,58,860                                  ?         1.835886e+07   
3        30,00,000                                  ?         3.000000e+06   
4        18,00,000                                  ?         1.800000e+06   
...            ...                                ...                  ...   
3039     45,00,000                                  ?         4.500000e+06   
3040      8,25,000                Govt backed VC Fund         8.250000e+05   
3041     15,00,000                                  ?         1.500000e+06   
3042             ?  Strategic Funding, Minority stake         1.840230e+07   
3043      1,40,000                                  ?         1.400000e+05   

           Date  
0    2020-01-09  
1    2020-01-13  
2    2020-01-09  
3    2020-01-02  
4    2020-01-02  
...         ...  
3039 2015-01-29  
3040 2015-01-29  
3041 2015-01-30  
3042 2015-01-30  
3043 2015-01-31  

[3044 rows x 12 columns]

Now in the dataset, we can see that missing values are denoted in "?". So the first step is to convert them into proper missing values, and then decide how to handle them. We use **df.replace()**. 

In [23]:
df = df.replace("?", np.nan)

In [24]:
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039                  NaN                                    NaN   
3040                  NaN                                    NaN   
3041                  NaN                                    NaN   
3042                  NaN                                    NaN   
3043                  NaN                                    NaN   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039            NaN        Asia Pacific Internet Group        Private Equity   
3040            NaN                     KARSEMVEN Fund        Private Equity   
3041            NaN     Exfinity Fund, GrowX Ventures.        Private Equity   
3042            NaN                         MakeMyTrip        Private Equity   
3043            NaN  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks  Amount_in_USD_Clean  \
0     20,00,00,000                                NaN         2.000000e+08   
1        80,48,394                                NaN         8.048394e+06   
2      1,83,58,860                                NaN         1.835886e+07   
3        30,00,000                                NaN         3.000000e+06   
4        18,00,000                                NaN         1.800000e+06   
...            ...                                ...                  ...   
3039     45,00,000                                NaN         4.500000e+06   
3040      8,25,000                Govt backed VC Fund         8.250000e+05   
3041     15,00,000                                NaN         1.500000e+06   
3042           NaN  Strategic Funding, Minority stake         1.840230e+07   
3043      1,40,000                                NaN         1.400000e+05   

           Date  
0    2020-01-09  
1    2020-01-13  
2    2020-01-09  
3    2020-01-02  
4    2020-01-02  
...         ...  
3039 2015-01-29  
3040 2015-01-29  
3041 2015-01-30  
3042 2015-01-30  
3043 2015-01-31  

[3044 rows x 12 columns]

Let us check how many missing values are there using **.isnull()**. 

In [25]:
df.isnull().sum()

Sr_No                     0
Date_dd/mm/yyyy           0
Startup_Name              0
Industry_Vertical       171
SubVertical             936
City__Location          180
Investors_Name           24
InvestmentnType           4
Amount_in_USD           964
Remarks                2625
Amount_in_USD_Clean       0
Date                      7
dtype: int64

So we fill the missing values in the columns "Industry_Vertical", "SubVertical", "City__Location", "Investors_Name", "InvestmentnType" and "Remarks" with "No Mention". Also missing values in "Date" column with "2015-01-01". 

In [26]:
df["Date"] = df["Date"].fillna(pd.Timestamp("2015-01-01"))

In [27]:
Columns = ["Industry_Vertical", "SubVertical", "City__Location", "Investors_Name", "InvestmentnType", "Remarks"]  # to replace multiple columns in a single step
df[Columns] = df[Columns].fillna("No Mention")

Now let us see how it went. 

In [28]:
df.isnull().sum()

Sr_No                    0
Date_dd/mm/yyyy          0
Startup_Name             0
Industry_Vertical        0
SubVertical              0
City__Location           0
Investors_Name           0
InvestmentnType          0
Amount_in_USD          964
Remarks                  0
Amount_in_USD_Clean      0
Date                     0
dtype: int64

Here you can see that we replaced missing values succsssfully. Note that we have an alternate column for "Amount_in_USD", which is "Amount_in_USD_Clean". 

### 2.4 Creating Derived Columns

We can create a column "Category" based on "Amount_in_USD_Clean", Which separates each company into "Tier-1", "Tier-2" and "Tier-3". First we check whats the maximum and minimum value in "Amount_in_USD_Clean". 

In [29]:
max_amnt = df["Amount_in_USD_Clean"].max()  # Maximum Amount
min_amnt = df["Amount_in_USD_Clean"].min()  # Minimum Amount

print("Maximum Amount:", max_amnt)
print("Minimum Amount:", min_amnt)

Maximum Amount: 3900000000.0
Minimum Amount: 16000.0


Therefore, 
- Tier-1: >=2000000000
- Tier-2: >=1000000000
- Tier-3: else

In [30]:
def categorize(amount):                 # create a function
    if amount >= 2000000000:
        return "Tier-1"
    elif amount >= 1000000000:
        return "Tier-2"
    else:
        return "Tier-3"

df["Category"] = df["Amount_in_USD_Clean"].apply(categorize)   # applies the function to each row of the "Amount_in_USD_Clean" column. 

In [31]:
df

Sr_No Date_dd/mm/yyyy                  Startup_Name  \
0         1      09-01-2020                      BYJU’S   
1         2      13-01-2020                        Shuttl   
2         3      09-01-2020                     Mamaearth   
3         4      02-01-2020  https://www.wealthbucket.in/   
4         5      02-01-2020                        Fashor   
...     ...             ...                           ...   
3039   3040      29-01-2015                    Printvenue   
3040   3041      29-01-2015                      Graphene   
3041   3042      30-01-2015                Mad Street Den   
3042   3043      30-01-2015                     Simplotel   
3043   3044      31-01-2015              couponmachine.in   

        Industry_Vertical                            SubVertical  \
0                  E-Tech                             E-learning   
1          Transportation              App based shuttle service   
2              E-commerce  Retailer of baby and toddler products   
3                 FinTech                      Online Investment   
4     Fashion and Apparel            Embroiled Clothes For Women   
...                   ...                                    ...   
3039           No Mention                             No Mention   
3040           No Mention                             No Mention   
3041           No Mention                             No Mention   
3042           No Mention                             No Mention   
3043           No Mention                             No Mention   

     City__Location                     Investors_Name       InvestmentnType  \
0         Bengaluru            Tiger Global Management  Private Equity Round   
1           Gurgaon          Susquehanna Growth Equity              Series C   
2         Bengaluru              Sequoia Capital India              Series B   
3         New Delhi                     Vinod Khatumal          Pre-series A   
4            Mumbai            Sprout Venture Partners            Seed Round   
...             ...                                ...                   ...   
3039     No Mention        Asia Pacific Internet Group        Private Equity   
3040     No Mention                     KARSEMVEN Fund        Private Equity   
3041     No Mention     Exfinity Fund, GrowX Ventures.        Private Equity   
3042     No Mention                         MakeMyTrip        Private Equity   
3043     No Mention  UK based Group of Angel Investors          Seed Funding   

     Amount_in_USD                            Remarks  Amount_in_USD_Clean  \
0     20,00,00,000                         No Mention         2.000000e+08   
1        80,48,394                         No Mention         8.048394e+06   
2      1,83,58,860                         No Mention         1.835886e+07   
3        30,00,000                         No Mention         3.000000e+06   
4        18,00,000                         No Mention         1.800000e+06   
...            ...                                ...                  ...   
3039     45,00,000                         No Mention         4.500000e+06   
3040      8,25,000                Govt backed VC Fund         8.250000e+05   
3041     15,00,000                         No Mention         1.500000e+06   
3042           NaN  Strategic Funding, Minority stake         1.840230e+07   
3043      1,40,000                         No Mention         1.400000e+05   

           Date Category  
0    2020-01-09   Tier-3  
1    2020-01-13   Tier-3  
2    2020-01-09   Tier-3  
3    2020-01-02   Tier-3  
4    2020-01-02   Tier-3  
...         ...      ...  
3039 2015-01-29   Tier-3  
3040 2015-01-29   Tier-3  
3041 2015-01-30   Tier-3  
3042 2015-01-30   Tier-3  
3043 2015-01-31   Tier-3  

[3044 rows x 13 columns]

### 2.5 Filtering or Aggregating Data

**2.5.(a) Filtering based on"Category"**

Filter rows where "Category" is "Tier-2". 

In [32]:
filtered_t2 = df[df["Category"] == "Tier-2"]
filtered_t2

Sr_No Date_dd/mm/yyyy Startup_Name Industry_Vertical  \
31      32      25-11-2019        Paytm           FinTech   
830    831      18-05-2017        Paytm         ECommerce   
966    967      21-03-2017     Flipkart         eCommerce   

                            SubVertical City__Location  \
31                        Mobile Wallet          Noida   
830  Mobile Wallet & ECommerce platform      Bangalore   
966               ECommerce Marketplace      Bangalore   

                        Investors_Name InvestmentnType   Amount_in_USD  \
31                Vijay Shekhar Sharma   Funding Round  1,00,00,00,000   
830                     SoftBank Group  Private Equity  1,40,00,00,000   
966  Microsoft, eBay, Tencent Holdings  Private Equity  1,40,00,00,000   

        Remarks  Amount_in_USD_Clean       Date Category  
31   No Mention         1.000000e+09 2019-11-25   Tier-2  
830  No Mention         1.400000e+09 2017-05-18   Tier-2  
966  No Mention         1.400000e+09 2017-03-21   Tier-2

**2.5.(b) Aggregation**

Next we calculate the sum of "Amount_in_USD_Clean" by grouping the "Category" column. 

In [33]:
agg_sum = df.groupby("Category")["Amount_in_USD_Clean"].sum().reset_index()
agg_sum

Category  Amount_in_USD_Clean
0   Tier-1         6.400000e+09
1   Tier-2         3.800000e+09
2   Tier-3         4.581661e+10